### PR TITLE
Updates to tedium for app-storage elements

### PR DESCRIPTION
**Pull request by rictic**
/cc @cdata @e111077

### CODE REVIEW

File: cleanup-passes/license-header.ts [hunk 1]
```diff
@@ -98,7 +98,12 @@ async function addLicenseHeader(element: ElementRepo): Promise<void> {
     const isHTML = path.extname(filename) === '.html';
     const startToken = isHTML ? /<!--/ : /\/[*]{1,2}/;
     const endToken = isHTML ? /-->/ : /\*\//;
-    let content = fs.readFileSync(realFilePath, 'utf-8');
```
Comment (garlicnation) [inline cleanup-passes/license-header.ts]: Optional: `await promisify(fs.readFile)(realFilePath, 'utf-8')`

**Comment by dfreedm** [comment]
LGTM

File: cleanup-passes/license-header.ts [hunk 2]
```diff
@@ -98,7 +98,12 @@ async function addLicenseHeader(element: ElementRepo): Promise<void> {
...
+    let content: string;
+    try {
+      content = fs.readFileSync(realFilePath, 'utf-8');
+    } catch(e) {
+      continue; // skip failure to read symlink into bower_components dir
```
Comment (justinfagnani) [inline cleanup-passes/license-header.ts]: can you comment why we might see a symlink, and why it's ok to skip?

File: tedium.ts [hunk 3]
```diff
@@ -157,20 +157,18 @@ async function getRepos(): Promise<GitHub.Repo[]> {
...
-  progressBar.tick();
-  const repos = [repo];
+  const repoPromises = [promisify(github.repos.get)({user: 'Polymer', repo: 'polymer'})];
+
   if (opts.repo.length) {
     // cleanup passes wants ContributionGuide around
-    repos.push(
-        await promisify(github.repos.get)(
+    repoPromises.push(
+        promisify(github.repos.get)(
```
Comment (justinfagnani) [inline tedium.ts]: you're promisifying multiple times, maybe store it in a variable

File: tedium.ts [hunk 4]
```diff
@@ -189,6 +187,8 @@ async function getRepos(): Promise<GitHub.Repo[]> {
     progressBar.tick();
   }
 
+  repos.push(...await Promise.all(repoPromises))
```
Comment (justinfagnani) [inline tedium.ts:189]: 👍 (on awaiting in parallel)

File: tedium.ts [hunk 5]
```diff
@@ -386,11 +386,36 @@ function connectToGithub() {
...
-  const dirs = fs.readdirSync('repos/');
+  const dirsToConsider: string[] = [];
   const htmlFiles: string[] = [];
 
-  for (const dir of dirs) {
+  for (const dir of fs.readdirSync('repos/')) {
+    dirsToConsider.push(path.join('repos', dir));
     for (const fn of fs.readdirSync(path.join('repos', dir))) {
+      try {
+        const stat = await promisify(fs.stat)(path.join('repos', dir, fn));
```
Comment (justinfagnani) [inline tedium.ts]: directly awaiting an async fs call doesn't seem to have any advantages over using the sync call...

File: tedium.ts [hunk 6]
```diff
@@ -386,11 +386,36 @@ function connectToGithub() {
...
  */
 async function analyzeRepos() {
-  const dirs = fs.readdirSync('repos/');
+  const dirsToConsider: string[] = [];
   const htmlFiles: string[] = [];
 
-  for (const dir of dirs) {
+  for (const dir of fs.readdirSync('repos/')) {
+    dirsToConsider.push(path.join('repos', dir));
     for (const fn of fs.readdirSync(path.join('repos', dir))) {
```
Comment (justinfagnani) [inline tedium.ts]: what's the name `fn` mean?

**Comment by justinfagnani** [comment]
Can you say a little about how this relates to the storage elements? Is it because they have new patterns in their project layout?

File: tedium.ts (hunk 5 above)
Comment (garlicnation) [inline tedium.ts]: Other work can continue during the async call - I believe the VM is blocked during the sync call

Comment (justinfagnani) [inline tedium.ts]: there doesn't seem to be any other work enqueued at this point

File: tedium.ts [hunk 7]
```diff
@@ -386,11 +386,36 @@ function connectToGithub() {
...
+        'test', 'util', 'explainer', 'src', 'helpers', 'site', 'templates',
+        'viewer', 'demo', 'patterns'
+      ]);
+      if (dirnamesToIgnore.has(path.basename(fn))) {
+        continue;
+      }
+      dirsToConsider.push(path.join('repos', dir, fn));
+    }
+  }
+
```
Comment (justinfagnani) [inline tedium.ts:415]: medium term nit: long file is long... it'd be really nice to break this up

File: cleanup-passes/license-header.ts (hunk 2 above)
Comment (rictic) [inline cleanup-passes/license-header.ts]: Added more comment.

File: tedium.ts (hunk 3 above)
Comment (rictic) [inline tedium.ts]: Done. Cleaned up a little in general here.

File: tedium.ts (hunk 5 above)
Comment (rictic) [inline tedium.ts]: Good point. Changed.

File: tedium.ts (hunk 6 above)
Comment (rictic) [inline tedium.ts]: Renamed to filename

**Comment by rictic** [comment]
Yeah, the newly public elements that are part of the app-storage group have some new filesystem patterns that were crashing tedium and/or making it generate unhelpful README files.

**Comment by garlicnation** [comment]
@rictic Put that in the commit message?

**Comment by rictic** [comment]
+1, I'll add it into the message when I squash and merge

**Comment by justinfagnani** [comment]
checked in symlinks? hmm... https://github.com/PolymerElements/app-pouchdb/issues/11

File: config.json [hunk 8]
```diff
@@ -11,6 +11,7 @@
         "repos/neon-elements",
         "repos/gold-elements",
         "repos/platinum-elements",
+        "repos/carbon-elements",
```
Comment (justinfagnani) [inline config.json:14]: I thought carbon- was no more?

File: config.json [hunk 9]
```diff
@@ -35,6 +36,18 @@
...
+    },
+    "issue-template": {
+      "blacklist": [
+        "repos/molecules",
+        "repos/iron-elements",
+        "repos/paper-elements",
+        "repos/neon-elements",
+        "repos/gold-elements",
+        "repos/platinum-elements",
+        "repos/carbon-elements",
```
Comment (justinfagnani) [inline config.json:48]: remove?

File: config.json (hunk 8 above)
Comment (rictic) [inline config.json:14]: No harm in leaving this in the blacklist

File: config.json (hunk 9 above)
Comment (rictic) [inline config.json:48]: No harm in leaving this in the blacklist

**Comment by justinfagnani** [comment]
LGTM

**Pull request merged**